cheroot/server.py

```diff
@@ -1175,6 +1175,11 @@ def send_headers(self):
                 if not self.close_connection:
                     self.outheaders.append((b'Connection', b'Keep-Alive'))
 
+        self.outheaders.append((
+            b'Keep-Alive',
+            'timeout={}'.format(self.server.timeout).encode('ISO-8859-1'),
+        ))
```

Comment:
According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Keep-Alive, `Keep-Alive` only has meaning if there's `Connection: Keep-Alive` set.

This is also confirmed by https://tools.ietf.org/html/rfc2068#section-19.7.1.1:
>   If the Keep-Alive header is sent, the corresponding connection token
>   MUST be transmitted. The Keep-Alive header MUST be ignored if
>   received without the connection token.

So this should be done under the condition that `connection` is set to `keep-alive`.

Reply:
@webknjaz I've adjusted code and tests accordingly :+1: